tasks/build/download_node_builds.js

```diff
@@ -1,84 +1,126 @@
     });
+  };
 
-    // use an async iife to store promise for download
-    // then store platform in active downloads list
-    // which we will read from in the finish task
-    platform.downloadPromise = (async function () {
-      grunt.file.mkdir(downloadDir);
+  const checkShaSum = (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
+    const expected = {
+      hash: 'sha256',
+      expected: platform.win ? shaSums[basename(dirname(platform.nodeUrl)) + '/' + file] : shaSums[file]
+    };
 
-      if (platform.win) {
-        await fromNode(cb => {
-          resp
-          .pipe(createWriteStream(resolve(downloadDir, 'node.exe')))
-          .on('error', cb)
-          .on('finish', cb);
-        });
+    if (!grunt.file.isDir(platform.nodeDir)) {
+      return false;
+    }
+
+    return checkHashFromFileAsync(path, expected).then(([passed, actual]) => {
+      if (passed) {
+        grunt.log.ok(`${platform.name} shasum verified`);
       } else {
-        await fromNode(cb => {
-          resp
-          .pipe(createGunzip())
-          .on('error', cb)
-          .pipe(new Extract({ path: downloadDir, strip: 1 }))
-          .on('error', cb)
-          .on('end', cb);
-        });
+        grunt.log.error(`${platform.name} shasum check failed`);
       }
+      return passed;
+    });
+  };
 
-      await fromNode(cb => {
-        rename(downloadDir, finalDir, cb);
-      });
-    }());
+  const getNodeBuild = (platform) => {
+    const downloadDir = platform.nodeDir;
+    const file = basename(platform.nodeUrl);
+
+    if (grunt.file.isDir(platform.nodeDir)) {
+      grunt.file.delete(platform.nodeDir);
+    }
 
-    activeDownloads.push(platform);
+    grunt.log.ok(`downloading ${platform.name}`);
+
+    return wreckGetAsync(platform.nodeUrl)
+    .then(([resp, payload]) => {
+      if (resp.statusCode !== 200) {
+        throw new Error(`${platform.nodeUrl} failed with a ${resp.statusCode} response`);
+      }
+      grunt.file.mkdir(downloadDir);
+      return payload;
+    })
+    .then(payload => writeFileAsync(resolve(downloadDir, file), payload));
 
-    var bytes = parseInt(resp.headers['content-length'], 10) || 'unknown number of';
-    var mb = ((bytes / 1024) / 1024).toFixed(2);
-    grunt.log.ok(`downloading ${platform.name} - ${mb} mb`);
   };
 
-  grunt.registerTask('_build:downloadNodeBuilds:start', function () {
-    map(platforms, start).nodeify(this.async());
-  });
+  const extractNodeBuild = async (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
 
-  grunt.registerTask('_build:downloadNodeBuilds:finish', function () {
-    map(activeDownloads, async (platform) => {
-      await platform.downloadPromise;
-      grunt.log.ok(`${platform.name} download complete`);
-    })
-    .nodeify(this.async());
+    return new Promise((resolve, reject) => {
```

Comment:
that was intentional, leaving archives in place, to prevent redownloading on each run. 
(if we remove archive we cannot be sure that the extracted binaries are valid on the next run)
you say this archives land in the final build ... should i just store them elsewhere ?